src/gui/guiScrollBar.cpp

```diff
@@ -183,36 +183,41 @@ void GUIScrollBar::draw()
 			slider_rect.LowerRightCorner.Y =
 					slider_rect.UpperLeftCorner.Y + thumb_size;
 		}
-		skin->draw3DButtonPaneStandard(this, slider_rect, &AbsoluteClippingRect);
+		
+		// Clip scrollbar so it doesn't show behind buttons for elastic scrolling
+		if (up_button->isVisible() || down_button->isVisible()) {
+            core::rect<s32> clip = AbsoluteClippingRect;
+            if (up_button->isVisible())
+                clip.UpperLeftCorner.Y = up_button->getAbsolutePosition().LowerRightCorner.Y;
+            if (down_button->isVisible())
+                clip.LowerRightCorner.Y = down_button->getAbsolutePosition().UpperLeftCorner.Y;
```

Comment:
What about horizontal scrollbars?

Reply:
Unfinished but easy to put in. I believe I mentioned fixing it when someone would properly review this, it just takes a few lines to fix. At the most I wanted the new contentdb scrollbars to look good so I will need to do it soon before that bug gets fixed (yes, exploiting a bug to fix a bug)